localstack/services/sqs/provider.py

```diff
@@ -1233,6 +1233,10 @@ def receive_message(
                     message_group_id=standard_message.message_group_id,
                 )
 
+                if isinstance(queue, FifoQueue):
+                    message_group = queue.get_message_group(standard_message.message_group_id)
+                    queue.update_message_group_visibility(message_group)
```

Comment:
i really regret not designing this in a way that we can easily do DLQ processing in the SQS abstraction, and have to do this on API level. well :shrug: